File: ClientApp/src/app/models/discussion/question.ts
export interface IQuestion {
    id?: string;
    elementId?: string;
    userId: string;
    title: string;
    text: string;
    creationDate?: string;
    lastEditDate?: string;
}

export interface IAnswer {
    id?: string;
    questionId?: string;
    userId?: string;
    text?: string;
    creationDate?: string;
    lastEditDate?: string;
}

File: Api.Entities/Models/Course/Discussion/Question.cs
namespace Api.Entities.Discussion
{
    public class Question : BaseEntity
    {
        public int ElementId { get; set; }
        public int UserId { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime LastEditDate { get; set; }
    }
}
